### PR TITLE
Changing the queued element sent to the Solver to allow different

### PR DESCRIPTION
**Pull request by albertshau**
types of solver tasks to allow solving for cluster creation or
addition of services or any other type of cluster operation that
requires solver interaction. As part of this, renaming
ClusterRequest to ClusterCreateRequest.

### CODE REVIEW

File: server/src/main/java/com/continuuity/loom/scheduler/SolverRequest.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+import com.google.common.base.Objects;
+
+/**
+ * Wrapper around different types of cluster requests that require solver interaction. Used so that we can pass
+ * different types of solver tasks on the same queue.
+ */
+public class SolverRequest {
+  private final Type type;
+  private final String serializedRequest;
```
Comment (poornachandra) [inline server/src/main/java/com/continuuity/loom/scheduler/SolverRequest.java]: Maybe call it jsonRequest for more clarity.

File: server/src/main/java/com/continuuity/loom/scheduler/SolverScheduler.java [hunk 2]
```diff
@@ -122,74 +126,86 @@ public String call() {
...
-          } catch (IllegalArgumentException e) {
-            LOG.error("Layout solving failed due to impossible constraints.", e);
-            errorMessage = errorMessage + ": " + e.getMessage();
+          SolverRequest solverRequest = GSON.fromJson(solveElement.getValue(), SolverRequest.class);
+          switch (solverRequest.getType()) {
+            case CREATE_CLUSTER:
+              return solveClusterCreate(cluster, GSON.fromJson(solverRequest.getSerializedRequest(),
+                                                               ClusterCreateRequest.class));
+            default:
+              return "unknown solver request type.";
```
Comment (poornachandra) [inline server/src/main/java/com/continuuity/loom/scheduler/SolverScheduler.java]: Returning the unknown solver request type will help in debugging.

**Comment by poornachandra** [comment]
Minor comments, otherwise LGTM

File: server/src/main/java/com/continuuity/loom/scheduler/SolverRequest.java (hunk 1 above)
Comment (albertshau) [inline server/src/main/java/com/continuuity/loom/scheduler/SolverRequest.java]: done

File: server/src/main/java/com/continuuity/loom/scheduler/SolverScheduler.java (hunk 2 above)
Comment (albertshau) [inline server/src/main/java/com/continuuity/loom/scheduler/SolverScheduler.java]: done

**Pull request merged**